bridge/qb/client/functions.lua

```diff
@@ -86,56 +121,134 @@ end
+
+    ---@type integer | {id: integer} | {id: integer, type: string, name: string}, vector3, number
+    local bone, coords, distance
+    for _, element in pairs(list) do
+        local boneCoords = GetWorldPositionOfEntityBone(entity, element.id or element)
+        local boneDistance = #(playerCoords - boneCoords)
+        if not coords or distance > boneDistance then
+            bone = element
+            coords = boneCoords
+            distance = boneDistance
+        end
+    end
+    if not bone then
+        bone = {id = GetEntityBoneIndexByName(entity, 'bodyshell'), type = 'remains', name = 'bodyshell'}
+        coords = GetWorldPositionOfEntityBone(entity, bone.id)
+        distance = #(coords - playerCoords)
+    end
+    return bone, coords, distance
+end
 
----@deprecated use GetBoneDistance from imports/utils.lua
-functions.GetBoneDistance = GetBoneDistance
+---@deprecated use the GetWorldPositionOfEntityBone native and calculate distance directly
+functions.GetBoneDistance = function(entity, boneType, bone)
+    local boneIndex = boneType == 1 and GetPedBoneIndex(entity, bone --[[@as integer]]) or GetEntityBoneIndexByName(entity, bone --[[@as string]])
+    local boneCoords = GetWorldPositionOfEntityBone(entity, boneIndex)
+    local playerCoords = GetEntityCoords(cache.ped)
+    return #(playerCoords - boneCoords)
+end
 
----@deprecated use AttachProp from imports/utils.lua
-functions.AttachProp = AttachProp
+---@deprecated use the AttachEntityToEntity native directly
+functions.AttachProp = function(ped, model, boneId, x, y, z, xR, yR, zR, vertex)
+    local modelHash = type(model) == 'string' and joaat(model) or model
+    local bone = GetPedBoneIndex(ped, boneId)
+    lib.requestModel(modelHash)
+    local prop = CreateObject(modelHash, 1.0, 1.0, 1.0, true, true, false)
+    AttachEntityToEntity(prop, ped, bone, x, y, z, xR, yR, zR, true, true, false, true, not vertex and 2 or 0, true)
+    SetModelAsNoLongerNeeded(modelHash)
+    return prop
+end
 
 -- Vehicle
 
----@deprecated call server function CreateVehicle instead from imports/utils.lua.
+---@deprecated use qbx.spawnVehicle from modules/lib.lua
 ---@param model string|number
 ---@param cb? fun(vehicle: number)
 ---@param coords? vector4 player position if not specified
 ---@param isnetworked? boolean defaults to true
 ---@param teleportInto boolean teleport player to driver seat if true
 function functions.SpawnVehicle(model, cb, coords, isnetworked, teleportInto)
-    coords = type(coords) == 'table' and vec4(coords.x, coords.y, coords.z, coords.w or GetEntityHeading(cache.ped)) or coords or GetCoordsFromEntity(cache.ped)
+    local playerCoords = GetEntityCoords(cache.ped)
+    local combinedCoords = vec4(playerCoords.x, playerCoords.y, playerCoords.z, GetEntityHeading(cache.ped))
+    coords = type(coords) == 'table' and vec4(coords.x, coords.y, coords.z, coords.w or GetEntityHeading(cache.ped)) or coords or combinedCoords
```

Comment:
```suggestion
    coords = type(coords) == 'table' and vec4(coords.x, coords.y, coords.z, coords.w or combinedCoords.w) or coords or combinedCoords
```
I guess we can re-use the heading instead of calculating it again.